core/frontend/src/tile/map/MapLayerFormatRegistry.ts

```diff
@@ -30,7 +30,7 @@ export class MapLayerFormat {
 
   /** Allow a source of a specific to be validated before being attached as a map-layer.
   */
-  public static async validateSource(_url: string, _userName?: string, _password?: string, _ignoreCache?: boolean): Promise<MapLayerSourceValidation> { return { status: MapLayerSourceStatus.Valid }; }
+  public static async validateSource(_url: string, _userName?: string, _password?: string, _ignoreCache?: boolean, _accesKey?: MapLayerKey): Promise<MapLayerSourceValidation> { return { status: MapLayerSourceStatus.Valid }; }
```

Comment:
@pmconne Not sure why we initially decided to pass deconstructed MapLayerSource properties here, but adding one more parameter feels wrong.
Any suggestion to adjust the API without making it too confusing?

Reply:
The "url' parameter we see everywhere is in fact the **service endpoint** URL which means any query parameters get discarded no matter what, so if you pass:

https://server/service?customParam=value => https://server/service?<Service_official_parameters_appended_here>

I think this behavior is ok (although it is somewhat unexpected given the actual documentation).
That's why I suggest adding a `customParameters` property where the user would be free to pass any custom parameters (which of course should not conflict with the service standard parameters).  The idea of having a `isSecret` prop, is for case where user want to use custom parameter like username=password=123.

Reply:
It's `@public` so your options are limited. What changes did you want to make?

Reply:
@pmconne  I created `validateSourceObj`, does that make sense to you ?

Reply:
Your default implementation of `validateSourceObj` does not forward to potentially-overridden implementation of `validateSource`. This will break any existing `MapLayerFormat` not controlled by us.

Reply:
> Did you encounter a problem adding an overload of `validateSource`, motivating the `Obj` suffix?

I'm getting: `Duplicate function implementation.ts(2393)`
If there's a way to create an override, I will happily remove the `Obj` suffix.

Reply:
> > Did you encounter a problem adding an overload of `validateSource`, motivating the `Obj` suffix?
> 
> I'm getting: `Duplicate function implementation.ts(2393)` If there's a way to create an override, I will happily remove the `Obj` suffix.

ok I think I found how to achieve it.

Reply:
Something like that, yes. Mark the one that has an implementation and ugly signature is `@internal`. Make the new one take a single argument, not two (e.g., add `source: MapLayerSource` to `ValidateSourceOptions`). Nitpickery, but our convention is to use the suffix `Args` for function argument interface types, not `Options`.

Reply:
Problem with overrides is that it invalidate sub-classe implementations, forcing them to implements all overrides of the base class.

I can certainly apply the overrides strategy to `MapLayerFormatRegistry.validateSource` but dont see how I can achieve it for `MapLayerFormat.validateSource` without breaking the API.

Reply:
> Problem with overrides is that it invalidate sub-class

Ugh. Fair.
I'd still prefer the new method to take a single `Args` argument - that's how we avoid similar problems in future when we want to evolve an API without breaking/deprecating it. If you can tolerate more extreme nitpickery, I'd ditch the `Obj` from `validateObj` - that non-word suffix isn't adding any useful information.